File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.js
function solution(babbling) {
    var answer = 0;
    
    let talk = ["aya", "ye", "woo", "ma"];

    for (let i=0; i<babbling.length; i++) {
        for (let j=0; j<talk.length; j++) {
            // 같은 발음 연속으로 하면 break
            if (babbling[i].includes(talk[j].repeat(2)))
                break;
            
            // 해당 발음 떼어내고 띄어쓰기 넣어두기
            babbling[i] = babbling[i].split(talk[j]).join(" ");
        }
        
        // 띄어쓰기 떼어내고 ""(크기 할당X)넣기
        if (babbling[i].split(" ").join("").length === 0)
            answer++;
    }
    
    return answer;
}

File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<string> babbling) {
    int answer = 0;
    
    vector<string> talk = { "aya", "ye", "woo", "ma" };

    for(int i=0; i<babbling.size(); i++){
        int check = 0;
        for(int j=0; j<babbling[i].length(); j++){
            if(babbling[i].substr(j, 3) == talk[0] && check != 1){
                check = 1;
                j+=2;
            }
            else if(babbling[i].substr(j, 2) == talk[1] && check != 2){
                check = 2;
                j+=1;
            }
            else if(babbling[i].substr(j, 3) == talk[2] && check != 3){
                check = 3;
                j+=2;
            }
            else if(babbling[i].substr(j, 2) == talk[3] && check != 4){
                check = 4;
                j+=1;
            }
            else{
                check = 0;
                break;
            }
        }
        
        if(check != 0)
            answer++;
    }
    
    return answer;
}